backend/slackbot/handler.js

```diff
@@ -1,91 +1,231 @@
 'use strict';
```

Comment:
Overall comment: I would split up some of your very long processing functions into helper functions. This makes your code easier to read and also easier to write unit tests for (because you can test the helper functions alone without worrying about the HTTP requests in your testing environment).